src/test/java/pl/wilki/battleship/board/BoardTest.java

```diff
@@ -11,6 +11,69 @@
  */
 @Test
 public class BoardTest {
+
+  @DataProvider
+  public static Object[][] newDefaultBoardFieldsCheck() {
+    return new Object[][]{
+        {0, FieldState.WATER},
+        {99, FieldState.WATER},
+        {1, FieldState.WATER},
+        {98, FieldState.WATER},
+        {45, FieldState.WATER},
+        {55, FieldState.WATER},
+        {50, FieldState.WATER},
+        {51, FieldState.WATER},
+        {49, FieldState.WATER},
+        {10, FieldState.WATER},
+        {90, FieldState.WATER},
+    };
+  }
+
+  @DataProvider
+  public static Object[][] setFieldCheck() {
```

Comment:
same here, rename this method
Also, you could generate this data, not hard-code it. Simple data generating algorithm for data and you can check all possible combinations.